src/components/osc-unique/oscUnique.js

```diff
@@ -31,21 +31,41 @@ angular.module('openshiftCommonUI')
     return {
       restrict: 'A',
       scope: {
-        oscUnique: '='
+        oscUnique: '=',
+        oscUniqueDisabled: '='
       },
       require: 'ngModel',
       link: function($scope, $elem, $attrs, ctrl) {
         var list = [];
+        var disabled = false;
+        var isValid = true;
 
         $scope.$watchCollection('oscUnique', function(newVal) {
           list = _.isArray(newVal) ?
                     newVal :
                     _.keys(newVal);
         });
 
+        $scope.$watch('oscUniqueDisabled', function(newVal) {
+          disabled = newVal;
+          if(!disabled) {
+            ctrl.$setValidity('oscUnique', isValid);
+            return;
+          }
+          ctrl.$setValidity('oscUnique', disabled);
+        });
+
         ctrl.$parsers.unshift(function(value) {
+          isValid = !_.includes(list, value);
+
+          // if disabled, skip validation, return true
+          if(disabled) {
+            ctrl.$setValidity('oscUnique', disabled);
+            return disabled;
```

Comment:
Isn't this supposed to return the input value, not `true`?

https://docs.angularjs.org/api/ng/type/ngModel.NgModelController#$parsers

I think this can be

```js
ctrl.$parsers.unshift(function(value) {
  isValid = !_.includes(list, value);
  ctrl.$setValidity('oscUnique', $scope.oscUniqueDisabled || isValid);

  return value;
});
```

You could pull the `$setValidity` line into its own function above so it's not repeated.

```js
var isUnique = true;

var updateValiditity = function() {
  ctrl.$setValidity('oscUnique', $scope.oscUniqueDisabled || isUnique);
};
$scope.$watch('oscUniqueDisabled', updateValidity);

ctrl.$parsers.unshift(function(value) {
  isUnique = !_.includes(list, value);
  updateValidity();
  return value;
});
```